tmail-backend/guice/distributed/src/main/java/com/linagora/tmail/contact/RabbitMQEmailAddressContactModule.java

```diff
@@ -42,9 +42,10 @@ public RabbitMQEmailAddressContactConfiguration rabbitMQEmailAddressContactConfi
     @Singleton
     public RabbitMQConfiguration getMailQueueConfiguration(RabbitMQEmailAddressContactConfiguration rabbitMQEmailAddressContactConfiguration) {
         return RabbitMQConfiguration.builder()
-            .amqpUri(rabbitMQEmailAddressContactConfiguration.getAmqpUri())
-            .managementUri(rabbitMQEmailAddressContactConfiguration.getAmqpUri())
-            .managementCredentials(rabbitMQEmailAddressContactConfiguration.getManagementCredentials())
+            .amqpUri(rabbitMQEmailAddressContactConfiguration.amqpUri())
+            .managementUri(rabbitMQEmailAddressContactConfiguration.amqpUri())
+            .managementCredentials(rabbitMQEmailAddressContactConfiguration.managementCredentials())
+            .vhost(rabbitMQEmailAddressContactConfiguration.vhost())
```

Comment:
Can't the vhost be instead obtained from the amqp uri?

I would not be expecting code changes to solve this issue...

Reply:
Vhost has been extract from amqp uri. 
```java
public Optional<String> vhost() {
        String vhostPath = amqpUri.getPath();
        if (vhostPath.startsWith("/")) {
            return Optional.of(vhostPath.substring(1))
                .filter(value -> !value.isEmpty());
        }
        return Optional.empty();
    }
```